From b454e4bfe86ab2dd07a1e09df91d9de08f7bf65e Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Thu, 23 Jun 2011 12:41:37 -0300
Subject: [RHEL6 qemu-kvm PATCH 062/115] usb-linux: We only need to keep track of 15 endpoints

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1308832951-8995-62-git-send-email-kraxel@redhat.com>
Patchwork-id: 27902
O-Subject: [RHEL-6.2 kvm PATCH 061/115] usb-linux: We only need to keep track of 15 endpoints
Bugzilla: 561414 632299 645351 711354
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

From: Hans de Goede <hdegoede@redhat.com>

Currently we reserve room for endpoint data for 16 endpoints, but given
that we only use endpoint data for endpoints 1-15, and always index the
array with the endpoint-number - 1, 15 is enough.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
(cherry picked from commit 3a4854b372e0ca1a619d33531efa02477674e93f)
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
---
 usb-linux.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 usb-linux.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/usb-linux.c b/usb-linux.c
index c3547a9..a34eb6a 100644
--- a/usb-linux.c
+++ b/usb-linux.c
@@ -78,7 +78,7 @@ typedef int USBScanFunc(void *opaque, int bus_num, int addr, int devpath,
 
 #define USBPROCBUS_PATH "/proc/bus/usb"
 #define PRODUCT_NAME_SZ 32
-#define MAX_ENDPOINTS 16
+#define MAX_ENDPOINTS 15
 #define USBDEVBUS_PATH "/dev/bus/usb"
 #define USBSYSBUS_PATH "/sys/bus/usb"
 
@@ -719,7 +719,7 @@ static int usb_host_set_interface(USBHostDevice *s, int iface, int alt)
     struct usbdevfs_setinterface si;
     int i, ret;
 
-    for (i = 1; i < MAX_ENDPOINTS; i++) {
+    for (i = 1; i <= MAX_ENDPOINTS; i++) {
         if (is_isoc(s, i)) {
             usb_host_stop_n_free_iso(s, i);
         }
@@ -1250,7 +1250,7 @@ static int usb_host_close(USBHostDevice *dev)
 
     qemu_set_fd_handler(dev->fd, NULL, NULL, NULL);
     dev->closing = 1;
-    for (i = 1; i < MAX_ENDPOINTS; i++) {
+    for (i = 1; i <= MAX_ENDPOINTS; i++) {
         if (is_isoc(dev, i)) {
             usb_host_stop_n_free_iso(dev, i);
         }
-- 
1.7.3.2

